pkg/providers/tinkerbell/machinetemplate.go

```diff
@@ -0,0 +1,27 @@
+package tinkerbell
+
+import (
+	"context"
+
+	"github.com/pkg/errors"
+	tinkerbellv1 "github.com/tinkerbell/cluster-api-provider-tinkerbell/api/v1beta1"
+	"k8s.io/apimachinery/pkg/api/equality"
+
+	"github.com/aws/eks-anywhere/pkg/clients/kubernetes"
+)
+
+// GetMachineTemplate gets a TinkerbellMachineTemplate object using the provided client
+// If the object doesn't exist, it returns a NotFound error.
+func GetMachineTemplate(ctx context.Context, client kubernetes.Client, name, namespace string) (*tinkerbellv1.TinkerbellMachineTemplate, error) {
```

Comment:
Is this used externally to this package by anything?

Reply:
`WorkersSpec` would use it as well. https://github.com/aws/eks-anywhere/pull/4764#discussion_r1088339772